### PR TITLE
Docs: Update documentation for v2.0.0

### PR DESCRIPTION
**Pull request by CruzMolina**
# Task: Update documentation for v2.0.0

## High Level Description

Doc changes:
- run `docify` to update current contract docs & add new contract documentation (`CamelCaps`)
- run `generate:doc` to update control-flow & uml docs

Build changes:
- pin `solc` to `0.6.10` to resolve `solidity-docgen` versioning error
- add `sol2uml` as necessary devDependency for generating docs

Misc changes:
- fix: lowercase links for `SUMMARY.md` in docify script
- fix: remove call to lowercase doc filenames in docify script
- fix: update `doc:control-flow` & `doc:uml` script file paths in `package.json`
- refactor: `mv` docify script into `scripts/docs` folder

### Documentation

- [X] Is your code up to date with the spec? 
- [X] Have you added your tests to the testing doc?

### CODE REVIEW

**Review by aparnakr — APPROVED** [review]

File: package.json [hunk 1]
```diff
@@ -66,7 +66,7 @@
     "doc:uml:factory": "sol2uml ./contracts/OtokenFactory.sol -f png -o ./docs/uml/GammaFactory.png",
     "doc:uml:whitelist": "sol2uml ./contracts/Whitelist.sol -f png -o ./docs/uml/GammaWhitelist.png",
     "doc:uml:addressbook": "sol2uml ./contracts/AddressBook.sol -f png -o ./docs/uml/GammaAddressbook.png",
-    "docify": "node docify.js",
+    "docify": "node scripts/docify.js",
```
Comment (haythem96) [inline package.json:69]: I think we should find another folder for this script, as this folder meant for the scripts that help interacting with the smart contracts IMO.

**Review by haythem96 — APPROVED** [review]

File: package.json (hunk 1 above)
Comment (antoncoding) [reply package.json:69]: maybe update the folder structure in `scripts/`?

Comment (CruzMolina) [reply package.json:69]: Sounds good :+1:

Comment (CruzMolina) [reply package.json]: Done ✅

**Pull request merged**